gateway-ha/src/main/java/io/trino/gateway/ha/router/HaResourceGroupsManager.java

```diff
@@ -402,6 +369,11 @@ private ResourceGroupsGlobalPropertiesDao getDao(@Nullable String routingGroupDa
         return connectionManager.getJdbi(routingGroupDatabase).onDemand(ResourceGroupsGlobalPropertiesDao.class);
     }
 
+    private ExactMatchSourceSelectorsDao getExactMatchSourceSelectorsDao()
+    {
+        return connectionManager.getJdbi(null).onDemand(ExactMatchSourceSelectorsDao.class);
```

Comment:
We can initialize ExactMatchSourceSelectorsDao in constructor and remove this method:
```java
    private final ExactMatchSourceSelectorsDao exactMatchSourceSelectorsDao;

    public HaResourceGroupsManager(JdbcConnectionManager connectionManager)
    {
...
        this.exactMatchSourceSelectorsDao = connectionManager.getJdbi().onDemand(ExactMatchSourceSelectorsDao.class);
    }
```